diff -Naur old/multiselect/js/multiselect.js new/multiselect-v1.0/js/multiselect.js
--- old/multiselect/js/multiselect.js	2014-04-25 10:02:47.252814702 +0530
+++ new/multiselect-v1.0/js/multiselect.js	2014-04-25 10:02:33.220809053 +0530
@@ -254,14 +254,12 @@
 	}
     
     $.fn.multiselect = function( options ) {        
-        return this.each(function() {	
 			var $this = $(this),
 				data = $this.data();
 			
 			var settings = $.extend({}, $.multiselect.defaults, $.multiselect.setup, data, options);
 			
             return new Multiselect($(this), settings);
-        });  
     };
     
 })( jQuery );
\ No newline at end of file
