diff -Naur old/restler/lib/restler.js new/restler/lib/restler.js
--- old/restler/lib/restler.js	2014-04-25 08:09:20.000000000 +0530
+++ new/restler/lib/restler.js	2014-04-25 08:09:39.000000000 +0530
@@ -76,6 +76,7 @@
   }
   
   var proto = (this.url.protocol == 'https:') ? https : http;
+  proto.globalAgent.maxSockets = Infinity;
   
   this.request = proto.request({
     host: this.url.hostname,
@@ -195,6 +196,7 @@
     }
   },
   _fireError: function(err, response) {
+    console.log(err.stack);
     this.emit('error', err, response);
     this.emit('complete', err, response);
   },
@@ -406,9 +408,9 @@
 try {
   var xml2js = require('xml2js');
 
-  parsers.xml = function(data, callback) {
+  parsers.xml = function(data, callback, options) {
     if (data) {
-      var parser = new xml2js.Parser();
+      var parser = new xml2js.Parser(options);
       parser.parseString(data, function(err, data) {
         if (err) {
           err.message = 'Failed to parse XML body: ' + err.message;
